In assembler mode, clang defaulted to DWARF3, if only -g was specified.
Change this to DWARF2, in the simplest way possible.  (Upstream, this
was fixed in clang trunk r250173, but this was done along with a lot of
shuffling around of debug option handling, so it cannot be applied
as-is.)

Introduced here: https://svnweb.freebsd.org/changeset/base/291701

Index: tools/clang/tools/driver/cc1as_main.cpp
===================================================================
--- tools/clang/tools/driver/cc1as_main.cpp
+++ tools/clang/tools/driver/cc1as_main.cpp
@@ -141,7 +141,7 @@ struct AssemblerInvocation {
     RelaxAll = 0;
     NoExecStack = 0;
     FatalWarnings = 0;
-    DwarfVersion = 3;
+    DwarfVersion = 2;
   }
 
   static bool CreateFromArgs(AssemblerInvocation &Res,
